src/features/timerState/timerStateSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+export type TimerState = {
+    showTimer: boolean;
+    isPausedTimer: boolean;
+}
+
+const initialState = {
+    showTimer: false,
+    isPausedTimer: false
+} as TimerState
+
+const timerStateSlice = createSlice({
+    name: 'TimerSlice',
+    initialState,
+    reducers: {
+        toggleShowTimer(state) {
```

Comment:
I recommend not abusing the toggle approach for slice, as it requires a strict call order to reach the desired state — which in turn can be complicated due to multiple react re-renderers and complexity with useEffect dependency from different components